packages/electron-commons/src/node-entry.ts

```diff
@@ -1,78 +1,112 @@
+    // creating an access point at runtime application, so it could use the ENGINE_PARENT_ENV_ID to be able to retrieve all getRuntimeArguments values into the app while launching a new environment using the initializer provided from '@wixc3/engine-electron-node'
+    const rootEngineEnvHost = new BaseHost();
+    rootEngineEnvHost.name = ENGINE_ROOT_ENVIRONMENT_ID;
+    const rootCom = new Communication(rootEngineEnvHost, ENGINE_ROOT_ENVIRONMENT_ID);
+    rootCom.registerAPI<MetadataCollectionAPI>(metadataApiToken, {
+        getRuntimeArguments: () => command.runOptions,
+    });
 
-        const metadataProviderHost = new BaseHost();
-        metadataProviderHost.name = METADATA_PROVIDER_ENV_ID;
-        rootCom.registerEnv(METADATA_PROVIDER_ENV_ID, metadataProviderHost);
+    const metadataProviderHost = new BaseHost();
+    metadataProviderHost.name = METADATA_PROVIDER_ENV_ID;
+    rootCom.registerEnv(METADATA_PROVIDER_ENV_ID, metadataProviderHost);
 
-        config.push(
-            COM.use({
-                config: {
-                    connectedEnvironments: {
-                        [ENGINE_ROOT_ENVIRONMENT_ID]: {
-                            id: ENGINE_ROOT_ENVIRONMENT_ID,
-                            host: rootEngineEnvHost,
-                        },
-                        [METADATA_PROVIDER_ENV_ID]: {
-                            id: METADATA_PROVIDER_ENV_ID,
-                            host: metadataProviderHost,
-                        },
+    config.push(
+        COM.use({
+            config: {
+                connectedEnvironments: {
+                    [ENGINE_ROOT_ENVIRONMENT_ID]: {
+                        id: ENGINE_ROOT_ENVIRONMENT_ID,
+                        host: rootEngineEnvHost,
+                    },
+                    [METADATA_PROVIDER_ENV_ID]: {
+                        id: METADATA_PROVIDER_ENV_ID,
+                        host: metadataProviderHost,
                     },
                 },
-            })
-        );
+            },
+        })
+    );
+
+    const { close } = await runIPCEnvironment({
+        type: 'node',
+        name: environmentName,
+        bundlePath,
+        childEnvName: environmentContextName,
+        featureName,
+        config,
+        features,
+        options: runtimeOptions,
+        context: basePath,
+        parentEnvName,
+        env,
+    });
+
+    disposeEnv = close;
+}
+
+async function handleDisposeCommand() {
+    await Promise.all([disposeEnv?.()]);
+}
+
+const messageHandler = (message: unknown) => {
+    const command = message as NodeEnvironmentCommand;
+
+    switch (command.id) {
+        case 'nodeEnvironmentStartupCommand':
+            handleStartupCommand(command).catch((e) => {
+                emitEvent<NodeEnvironmentEvent>(process, {
+                    id: 'nodeEnvironmentInitFailedEvent',
+                    error: toError(e).message,
```

Comment:
this is a terrible practice. it looses the stack trace. avoid

Reply:
fixed, passing whole error object in message